<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <!-- The above 3 meta tags *must* come first in the head; any other head content must come *after* these tags -->
    <meta name="description" content="Escherichia coli Genetic Refference Panel for genotype to phenotype analysis">
    <meta name="author" content="Marco Galardini">
    <link rel="icon" href="../favicon.ico">

    <title>Ecoref: Escherichia coli genetic reference panel</title>

    <!-- Bootstrap core CSS -->
    <link href="../components/bootstrap/dist/css/bootstrap.min.css" rel="stylesheet">
    
    <!-- Custom styles -->
    <link href="../css/custom.css" rel="stylesheet">

    <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
      <script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
      <script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
    <![endif]-->
  </head>
  <body>
    <!-- Fixed navbar -->
    <nav class="navbar navbar-default navbar-fixed-top">
      <div class="container">
        <div class="navbar-header">
          <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false" aria-controls="navbar">
            <span class="sr-only">Toggle navigation</span>
          </button>
          <a class="navbar-brand" href="../">Ecoref</a>
        </div>
        <div id="navbar" class="navbar-collapse collapse">
          <ul class="nav navbar-nav navbar-right">
            <li><a href="../about">About</a></li>
            <li><a href="../strains">Strains</a></li>
            <li><a href="../conditions">Conditions</a></li>
            <li><a href="../variants">Variants</a></li>
            <li class="active"><a href="../download">Download</a></li>
            <li><a href="../contact">Contact</a></li>
          </ul>
        </div><!--/.nav-collapse -->
      </div>
    </nav>
    <div class="container">
      <div class="row">
        <div class="col-md-2 text-center">
          <span class="glyphicon glyphicon-info-sign glyphicon-bigger" aria-hidden="true"></span>
        </div>
        <div class="col-md-10">
          <h3>Data download terms and conditions</h3>
          <p>We encourage anyone to download and use the data provided here. If you do so, please do <a href="../contact/">get in touch with us</a>, we are happy to provide guidance in navigating this data.</p>
        </div>
      </div>
      <div class="row">
        <div class="col-md-2 text-center">
          <span class="glyphicon glyphicon-file glyphicon-bigger" aria-hidden="true"></span>
        </div>
        <div class="col-md-10">
          <h3>Citation</h3>
	  <p>Please cite the following paper if you publish any analysis involving the data deposited here</p>
          <p>
          <h4><a href="https://elifesciences.org/articles/31035">Phenotype inference in an Escherichia coli strain panel</a><br/>
          <small>eLife 2017;6:e31035; doi: <a href="https://doi.org/10.7554/eLife.31035">10.7554/eLife.31035</a></small></h4>
          </p>
        </div>
      </div>
      <div class="row">
        <div class="col-md-2 text-center">
          <span class="glyphicon glyphicon-download glyphicon-bigger" aria-hidden="true"></span>
        </div>
        <div class="col-md-10">
          <h3>Ecoref bulk data download (release 1)</h3>
          <ul>
            <li><a href="../data/strains.tsv" download>Strains details</a> (TSV format)</li>
            <li><a href="../data/conditions.tsv" download>Conditions details</a> (TSV format)</li>
            <li><a href="../data/phenotypic_data.tsv" download>Growth phenotypes</a></li>
            <li><a href="../data/snps.vcf.gz" download>SNPs</a> (in <a href="https://en.wikipedia.org/wiki/Variant_Call_Format">VCF format</a>) <span class="glyphicon glyphicon-exclamation-sign" data-toggle="tooltip" data-placement="top" title="Strains belonging to different species and duplicates have been excluded" aria-hidden="true"></span> <span class="glyphicon glyphicon-warning-sign" data-toggle="tooltip" data-placement="top" title="Due to the fact that variants were called through pairwise alignments with the reference, the dot notation in the VCF file can indicate either reference allele or missing region" aria-hidden="true"></span></li>
            <li><a href="../data/sift.tsv.gz" download>SIFT scores</a> for all the nonsynonymous substitutions observed in the strains collection <span class="glyphicon glyphicon-file" data-toggle="tooltip" data-placement="top" title="Format: Uniprot ID, nonsynonymous variant, SIFT p-value" aria-hidden="true"></span> <span class="glyphicon glyphicon-info-sign" data-toggle="tooltip" data-placement="top" title="A SIFT score below 0.05 indicates a deleterious variant" aria-hidden="true"></span></li>
            <li><a href="../data/foldx.tsv.gz" download>FoldX scores</a> for all the nonsynonymous substitutions observed in the strains collection <span class="glyphicon glyphicon-file" data-toggle="tooltip" data-placement="top" title="Format: Uniprot ID, nonsynonymous variant, FoldX ddG (kcal/mol)" aria-hidden="true"></span> <span class="glyphicon glyphicon-info-sign" data-toggle="tooltip" data-placement="top" title="A ddG higher than 2 kcal/mol indicates a deleterious variant" aria-hidden="true"></span></li>
            <li><a href="../data/disruption_scores.tsv.gz" download>gene disruption scores</a> for all K-12 coding genes in all strains<span class="glyphicon glyphicon-file" data-toggle="tooltip" data-placement="top" title="Format: rows are Uniprot ID, columns are strains" aria-hidden="true"></span> <span class="glyphicon glyphicon-info-sign" data-toggle="tooltip" data-placement="top" title="A disruption score above 0.9 might indicate a non-functional gene" aria-hidden="true"></span></li>
            <li><a href="../data/pangenome.csv.gz" download>Pangenome</a> (computed using <a href="https://sanger-pathogens.github.io/Roary/">Roary</a>) <span class="glyphicon glyphicon-exclamation-sign" data-toggle="tooltip" data-placement="top" title="Strains belonging to different species and duplicates have been excluded" aria-hidden="true"></span></li>
            <li><a href="../data/tree.nwk" download>Strains tree</a> (computed using <a href="http://harvest.readthedocs.io/en/latest/content/parsnp.html">Parsnp</a>) <span class="glyphicon glyphicon-exclamation-sign" data-toggle="tooltip" data-placement="top" title="Strains belonging to different species and duplicates have been excluded" aria-hidden="true"></span></li>
            <li><a href="http://www.ebi.ac.uk/~marco/annotated_assemblies.7z" download>All annotated assemblies</a> (in GFF3 format) <span class="glyphicon glyphicon-exclamation-sign" data-toggle="tooltip" data-placement="top" title="Strains belonging to different species and duplicates have been excluded" aria-hidden="true"></span></li>
            <li><a href="http://www.ebi.ac.uk/~marco/proteomes.tar.gz" download>All proteomes</a> (in fasta format) <span class="glyphicon glyphicon-exclamation-sign" data-toggle="tooltip" data-placement="top" title="Strains belonging to different species and duplicates have been excluded" aria-hidden="true"></span></li>
          </ul>
        </div>
      </div>
      <div class="row">
        <div class="col-md-2 text-center">
          <span class="glyphicon glyphicon-folder-open glyphicon-bigger" aria-hidden="true"></span>
        </div>
        <div class="col-md-10">
          <h3>Old releases</h3>
          <ul>
            <li>...</li>
          </ul>
        </div>
      </div>
      <div class="row">
        <div class="col-md-2 text-center">
          <span class="glyphicon glyphicon-console glyphicon-bigger" aria-hidden="true"></span>
        </div>
        <div class="col-md-10">
          <h3>How to read the pangenome table</h3>
          <p>The python <a href="http://pandas.pydata.org/">Pandas library</a> makes it relatively simple to load this data. You might want to look into <a href="https://github.com/sanger-pathogens/Roary/tree/master/contrib/roary_plots">roary_plots</a> to generate more complex plots.</p>
          <pre>import pandas as pd
import numpy as np

# Load roary
roary = pd.read_table('pangenome.csv',
                      sep=',',
                      low_memory=False)
# Set index (group name)
roary.set_index('Gene', inplace=True)
# Drop the other info columns
roary.drop(list(roary.columns[:13]), axis=1, inplace=True)

# Transform it in a presence/absence matrix (1/0)
roary.replace('.{2,100}', 1, regex=True, inplace=True)
roary.replace(np.nan, 0, regex=True, inplace=True)</pre>
        </div>
      </div>
    </div>
    <footer class="footer">
      <div class="container">
        <div class="navbar-header hidden-xs">
          <a href="http://www.ebi.ac.uk/research/beltrao"><img src="../img/ebi_logo.png"></a>
        </div>
        <div class="navbar-right">
          <p class="text-muted text-right">
            Copyright EMBL-EBI 2017
            </br>
            EBI is an outstation of the <a href="https://www.embl.de/">European Molecular Biology Laboratory</a>
            </br>
            <a href="http://www.ebi.ac.uk/about/terms-of-use">Terms of use</a>
          </p>
        </div>
      </div>
    </footer>
    <script src="../components/jquery/dist/jquery.min.js"></script>
    <script src="../components/bootstrap/dist/js/bootstrap.min.js"></script>
    <script type="text/javascript" charset="utf8" src="../js/download.js"></script>
    <script>
      (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
      })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

      ga('create', 'UA-91889624-1', 'auto');
      ga('send', 'pageview');
    </script>
  </body>
</html>
